examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts

```diff
@@ -136,3 +271,158 @@ export function addLanguage(patient: Patient, valueCoding: Coding, preferred: bo
+
+  await medplum.upsertResource(
+    {
+      resourceType: 'Coverage',
+      status: 'active',
+      beneficiary: createReference(patient),
+      subscriberId: subscriberId,
+      relationship: { coding: [answers['relationship-to-subscriber'].valueCoding as Coding] },
+      payor: [payor],
+    },
+    {
+      beneficiary: getReferenceString(patient),
+      payor: getReferenceString(payor),
+      // subscriberId: subscriberId,
+    }
+  );
+}
+
+export async function addConsent(
+  medplum: MedplumClient,
+  patient: Patient,
+  consentGiven: boolean,
+  scope: CodeableConcept,
+  category: CodeableConcept,
+  policyRule: CodeableConcept | undefined,
+  date: Consent['dateTime']
+): Promise<void> {
+  await medplum.createResource({
+    resourceType: 'Consent',
+    patient: createReference(patient),
+    status: consentGiven ? 'active' : 'rejected',
+    scope: scope,
+    category: [category],
+    policyRule: policyRule,
+    dateTime: date,
+  });
+}
+
+/**
+ * Finds a QuestionnaireItem or QuestionnaireResponseItem with the given linkId
+ *
+ * @param items - The array of objects present in the `.item` attribute of a Questionnaire or QuestionnaireResponse
+ * @param linkId - The id to be found
+ * @returns - The found item or undefined in case it's not found
+ */
+export function findQuestionnaireItem<T extends QuestionnaireItem | QuestionnaireResponseItem>(
+  items: T[] | undefined,
+  linkId: string
+): T | undefined {
+  if (!items) {
+    return undefined;
+  }
+
+  return items.reduce((foundItem: T | undefined, currentItem: T | undefined) => {
+    // If currentItem is undefined or the item was already found just return it
+    if (foundItem || !currentItem) {
+      return foundItem;
+    }
+
+    if (currentItem.linkId === linkId) {
+      return currentItem;
+    } else if (currentItem.item) {
+      // This enables traversing nested structures
+      return findQuestionnaireItem(currentItem.item as T[], linkId);
+    }
+
+    return undefined;
+  }, undefined);
+}
+
+/**
+ * Finds the answers to a group of items that can be repeated.
+ *
+ * @param questionnaire - The Questionnaire resource, used to find the list of linkIds that will compose each group
+ * @param response - The QuestionnaireResponse resource where the answers will be extracted
+ * @param groupLinkId - The linkId of the group that can be repeated in the questionnaire
+ * @returns - An array of objects each containing a set of grouped answers
+ */
+export function getGroupRepeatedAnswers(
+  questionnaire: Questionnaire,
+  response: QuestionnaireResponse,
+  groupLinkId: string
+): Record<string, QuestionnaireResponseItemAnswer>[] {
+  const questionnaireItem = findQuestionnaireItem(questionnaire.item, groupLinkId) as QuestionnaireItem;
+  const responseItem = findQuestionnaireItem(response.item, groupLinkId) as QuestionnaireResponseItem;
+
+  if (questionnaireItem.type !== 'group' || !questionnaireItem?.item) {
```

Comment:
```suggestion
  if (questionnaireItem.type !== 'group' ?? !questionnaireItem?.item) {
```

Prefer `??`

Reply:
@rahul1 I agree `??` is more adequate in `const responses = responseItem.item ?? [];` as it's a coalescing operator and semantically that's what we are doing in that expression. But in this case here I'd argue the "or" logical operator is more semantically correct. That said, I understand the end result is the same and I'm all in for following conventions over arguing over small details, LMK your final decision and I'll make the change here and in all the other places over this file.

Reply:
I'm sorry, you're right. I was reviewing a bit too quickly 😅. The `||` operator was correct